### ISSUE TITLE
Replacing or porting trackplacer

### ISSUE THREAD

**Issue by stevecotton** [wesnoth/wesnoth#4365]
The data/tools/traceplacer tool, in addition to being python2 (#1508) is written using the GTK2 toolkit which is also end-of-lifed:
* https://wiki.gnome.org/Projects/PyGObject/IntrospectionPorting#Porting_from_PyGTK_2_to_PyGI_GTK_3
* https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=885135#5

Of the 1275 lines in trackplacer, the first 358 handle the data model and file format, the remainder are the UI. Instead of porting or rewriting the GUI, I'm wondering if we could find an existing open-source tool that could edit map tracks.

Inkscape should be able to do it, but I can see it's going to have a steep learning curve (and I don't know how to do it).

Tiled seems a possibility, using its `objectlayer` functionality - before the weekend I'll push a WIP commit showing how to create TMX files for track placement. <https://www.mapeditor.org/>

**Comment by Elvish-Hunter** [comment]
> Of the 1275 lines in trackplacer, the first 358 handle the data model and file format, the remainder are the UI. Instead of porting or rewriting the GUI, I'm wondering if we could find an existing open-source tool that could edit map tracks.

The other option is to rewrite the GUI using Tkinter and the Pillow library (any other toolkit is going to require additional and heavy downloads to the users). However, the data model needs to be updated as well, because it tends to be quite confusing: for example, I'd use a Waypoint custom object to represent every item on the map, instead of tuples.

**Comment by stevecotton** [comment]
I'd be very happy for someone else to take on this ticket, either with or without the WIP that's auto-linked above.

Most of the mainline campaigns have been hand-edited after creation with trackplacer, and of the data will be lost when read by trackplacer (either the existing one or that WIP python3 port).

* NR's bigmap.cfg has a track RECOVERY whose STAGE1 starts with an OLD_REST, which will be ignored by trackplacer.
* LoW: the OLD_BATTLE of LoW's SAURIANS track will be lost.
* DM's SEARCH_STAGE1 is omitted from all subsequent parts of SEARCH. Also in DM, SEARCH_STAGE3 has a point which is then moved in STAGE4 onwards - I guess a hand edit. Both of this will be overwritten if the file is edited with trackplacer.
* SotA's journey_chapter*.cfg files have some of the trackplacer comments removed, they won't be handled by this tool, at least not until better error handling is added.
* WoV's bigmap.cfg file has some of the trackplacer comments removed.

What's in that WIP commit is a refactor of the trackplacer's .cfg file handling, deleting all the GUI code, porting to Python3 and including Elvish Hunter's suggestion of creating a Waypoint class. On the front of that is a command-line tool to export the data to Tiled's .tmx format.

The WIP commit can also read in a .cfg and then re-export it to a new .cfg file, to see if the data is preserved. The format is chosen by the output filename.

**Comment by AncientLich** [comment]
I like the idea of using tiled instead of trackplacer (optimizing resources is a good thing)... If I have a time I will take a look too into this possible direction... so the new 'trackplacer' tool could became a tool that converts .tmx to a .cfg file that should be imported in the WML scenario which will show the map path.

**Comment by stevecotton** [comment]
Now opened as PR #4464. It doesn't have an animation preview, but other than that I think it's reached feature-parity with trackplacer2.

**Comment by stevecotton** [comment]
For trackplacer, my new plan is to have the editing done in Tiled (the commit that's just merged), but have a standalone tool for checking the animations (without support for editing).
